### ISSUE TITLE
aws_ecs_task_definition does not detect changes to efs_volume_configuration block

### ISSUE THREAD

**Issue by jcarlson** [terraform-providers/terraform-provider-aws#12190]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform -v
Terraform v0.12.12
+ provider.aws v2.50.0
```

### Affected Resource(s)

* aws_ecs_task_definition

### Terraform Configuration Files

```hcl
# main.tf

provider "aws" {
  version = "~> 2.50.0"
  region  = "us-east-1"
}

resource "aws_ecs_task_definition" "main" {
  container_definitions    = file("${path.module}/container-definitions.json")
  family                   = "efs-test"
  network_mode             = "bridge"
  requires_compatibilities = ["EC2"]

  volume {
    name = "efs"

    efs_volume_configuration {
      file_system_id = "fs-12345678"
      root_directory = "/foo/bar"
    }
  }
}
```

```json
[
  {
    "cpu": 0,
    "essential": true,
    "image": "nginx",
    "memory": 2048,
    "memoryReservation": 1024,
    "mountPoints": [{
      "containerPath": "/data",
      "sourceVolume": "efs"
    }],
    "name": "nginx",
    "portMappings": [
      {
        "containerPort": 80,
        "hostPort": 0,
        "protocol": "tcp"
      }
    ],
    "ulimits": [
      {
        "name": "core",
        "softLimit": 0,
        "hardLimit": 0
      }
    ],
    "volumesFrom": []
  }
]
```

### Expected Behavior

Terraform should detect changes to the `efs_volume_configuration` block and update/replace the Task Definition accordingly.

### Actual Behavior

Changes to attributes in the `efs_volume_configuration` block are ignored and `terraform plan` returns clean.

### Steps to Reproduce

1. `terraform apply` the above configuration
2. modify the `root_directory` or `file_system_id` attributes of the `efs_volume_configuration` block
3. `terraform plan`

**Comment by jcarlson** [comment]
<img width="555" alt="image" src="https://user-images.githubusercontent.com/57132/75388273-ebf67d00-58a1-11ea-8aac-6382648d2ab3.png">

Also curious that my IDE (RubyMine 2019.3.3 with Terraform/HCL plugin 0.7.7) does not seem to recognize this block as being valid...

**Comment by ewbankkit** [comment]
It looks like `resourceAwsEcsTaskDefinitionVolumeHash()`

https://github.com/terraform-providers/terraform-provider-aws/blob/7744d9dac45f6a5df1dd9d5c3002a3bd3ec6c476/aws/resource_aws_ecs_task_definition.go#L558-L564

isn't considering `efs_volume_configuration` (or `docker_volume_configuration`).

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.64.0 of the Terraform AWS Provider, later this week. Thanks to @DrFaust92 for the implementation. 👍
